### ISSUE TITLE
[Bug]: Behavior of pixel_metrics/image_metrics

### ISSUE THREAD

**Issue by daniel-bogdoll** [openvinotoolkit/anomalib#2346]
### Describe the bug

When I setup my engine, I want to log all metrics available:

```
# https://anomalib.readthedocs.io/en/latest/markdown/guides/reference/metrics/index.html
engine = Engine(
                task=self.TASK,
                default_root_dir=self.models_path,
                logger=wandb_logger,
                max_epochs=max_epochs,
                callbacks=callbacks,
                pixel_metrics=[  
                    "AUPIMO",
                    "AUPR",
                    "AUPRO",
                    "AUROC",
                    "AnomalyScoreDistribution",
                    "BinaryPrecisionRecallCurve",
                    "F1AdaptiveThreshold",
                    "F1Max",
                    "F1Score",
                    "ManualThreshold",
                    "MinMax",
                    "PIMO",
                    "PRO",
                ],
                accelerator="auto",
            )
```

However, the Engine always loads the ```AnomalibMetricCollection```:

```
   ┃ Name                  ┃ Type                     ┃ Params ┃ Mode  ┃
┡━━━╇━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━╇━━━━━━━┩
│ 0 │ model                 │ DraemModel               │ 97.4 M │ train │
│ 1 │ loss                  │ DraemLoss                │      0 │ train │
│ 2 │ _transform            │ Resize                   │      0 │ train │
│ 3 │ normalization_metrics │ MetricCollection         │      0 │ train │
│ 4 │ image_threshold       │ F1AdaptiveThreshold      │      0 │ train │
│ 5 │ pixel_threshold       │ F1AdaptiveThreshold      │      0 │ train │
│ 6 │ image_metrics         │ AnomalibMetricCollection │      0 │ train │
│ 7 │ pixel_metrics         │ AnomalibMetricCollection │      0 │ train │
```

I do not understand why? The same behavior can be observed in the notebook below. Here, "AUROC" is selected, however ```AnomalibMetricCollection``` shows. I am experimenting with the ```Draem``` model. I am not so sure if this is a bug or if I do not understand some part of it, but after googling for an hour I am kind of stuck here. https://github.com/openvinotoolkit/anomalib/blob/404e896c769aa069f77deb79081e19024e8596a8/notebooks/600_loggers/601_mlflow_logging.ipynb#L1083

### Dataset

Other (please specify in the text field below)

### Model

DRAEM

### Steps to reproduce the behavior

See above

### OS information

OS information:
- OS: Ubuntu 24.04
- Python version: 3.12
- Anomalib version: recent
- PyTorch version: recent
- CUDA/cuDNN version: 12.4
- GPU models and configuration: RTX 4090
- Any other relevant information: Custom dataset in mvtec format. Simpler experiments with Padim worked.

### Expected behavior

My selected metrics should be chosen.

### Pip/GitHub

pip

### Configuration YAML

```yaml
---
```

### Logs

```shell
---
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by daniel-bogdoll** [comment]
Also, there is no information about the ```AnomalibMetricCollection``` in the docs:

![image](https://github.com/user-attachments/assets/6beb1825-7b1b-43f9-8ce8-f9390092bd61)

**Comment by blaz-r** [comment]
This is mostly an implementation detail. If you specify multiple metrics, they get stored inside a special object. This way you can update the values in all metrics simultaneously with a single `update` call and then compute them with a single `compute` call. 

For more details, you can check [AnomalibMetricCollection](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/metrics/collection.py) class, or it's parent class from pytorch lightning [MetricCollection](https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metriccollection).

I think that the docs don't show it, as it's not added to `__all__` inside [`__init__.py`](https://github.com/openvinotoolkit/anomalib/blob/404e896c769aa069f77deb79081e19024e8596a8/src/anomalib/metrics/__init__.py#L27-L41). @ashwinvaidya17 should this maybe be included?

**Comment by samet-akcay** [comment]
As @blaz-r pointed out, it is not shown since `AnomalibMetricCollection` is not within `__init__.py`. 
With that being said, the class does not have any specific implementational detail, but just wrapping the original Lightning MetricCollection class. One idea would be to expand its docstring with examples and a link to the Lightning documentation as they have comprehensive documentation 
https://lightning.ai/docs/torchmetrics/stable/pages/overview.html#metriccollection

I need to double check how the new metric design will impact the metric collection though
https://github.com/openvinotoolkit/anomalib/pull/2326

**Comment by daniel-bogdoll** [comment]
Thanks for the quick response, however I am not sure if I understand. When I log the training with wandb, the metrics I chose do not show up. wandb only shows the following ones:

- train_loss_step
- image_F1Score
- epoch
- pixel_F1Score
- pixel_AUROC
- image_AUROC
- train_loss_epoch

So I guess I assumed these were simply the metrics included in the ```AnomalibMetricCollection```, but that does not seem to be correct?

**Comment by blaz-r** [comment]
That is quite unusual. Are the metrics logged in the shell, but not on wandb? I think that the problem here is unrelated to AnomalibMetricCollection.

**Comment by daniel-bogdoll** [comment]
Works now, not exactly sure what I did :D Reinstallation helped.

**Comment by blaz-r** [comment]
Great
